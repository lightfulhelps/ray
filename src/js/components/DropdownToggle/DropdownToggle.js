// @flow
import * as React from 'react';
import classNames from 'classnames';
import { Button } from '../../';

type Props = {
  className?: string,
  isOpen?: boolean,
};

const DropdownToggle = ({ className, isOpen = false, ...other }: Props) => {
  const classes = classNames(className, 'dropdown-toggle');

  return (
    <Button
      icon={isOpen ? 'chevronUp' : 'chevronDown'}
      iconPosition="right"
      {...other}
      className={classes}
    />
  );
};

export default DropdownToggle;
