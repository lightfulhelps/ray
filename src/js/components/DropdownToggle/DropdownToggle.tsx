import * as React from 'react';
import classNames from 'classnames';
import { Button } from '../..';

type Props = {
  className?: string;
  isOpen?: boolean;
};

const DropdownToggle: React.FC<Props> = ({ className, isOpen = false, ...other }: Props) => {
  const classes = classNames(className, 'dropdown-toggle');

  return (
    <Button
      icon={isOpen ? 'caretUp' : 'caretDown'}
      iconPosition="right"
      isOutline
      theme="gray-600"
      {...other}
      className={classes}
    />
  );
};

export default DropdownToggle;
