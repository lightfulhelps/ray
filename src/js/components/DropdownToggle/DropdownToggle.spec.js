import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import DropdownToggle from './DropdownToggle';

const setup = (overrides = {}) => {
  const props = merge({}, overrides);
  const wrapper = shallow(<DropdownToggle {...props} />);

  return { wrapper, props };
};

describe('<DropdownToggle />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('dropdown-toggle')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle isOpen', () => {
    const { wrapper } = setup();

    expect(wrapper.prop('icon')).toEqual('caretDown');

    wrapper.setProps({ isOpen: true });

    expect(wrapper.prop('icon')).toEqual('caretUp');
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
