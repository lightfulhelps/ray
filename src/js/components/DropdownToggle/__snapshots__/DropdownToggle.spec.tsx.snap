// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownToggle /> should render 1`] = `
<Button
  className="dropdown-toggle"
  icon="chevronDown"
  iconPosition="right"
  isOutline={true}
  size="md"
  theme="primary"
/>
`;
